Fails to build with GHC 9.6.3

Hi @ddssff,

when building `haskell-unixutils` 1.54.2 on GHC 9.6.3, the following error occurs:
```
Configuring Unixutils-1.54.2...
Preprocessing library for Unixutils-1.54.2..
Building library for Unixutils-1.54.2..
[1 of 9] Compiling System.Unix.Chroot ( System/Unix/Chroot.hs, dist/build/System/Unix/Chroot.dyn_o )

System/Unix/Chroot.hs:48:27: error: [GHC-83865]
    • Couldn't match expected type: OpenFileFlags
                                    -> IO System.Posix.Types.Fd
                  with actual type: IO System.Posix.Types.Fd
    • The function ‘openFd’ is applied to four value arguments,
        but its type ‘FilePath
                      -> OpenMode -> OpenFileFlags -> IO System.Posix.Types.Fd’
        has only three
      In the second argument of ‘($)’, namely
        ‘openFd "/" ReadOnly Nothing defaultFileFlags’
      In a stmt of a 'do' block:
        rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
   |
48 |        rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

System/Unix/Chroot.hs:48:47: error: [GHC-83865]
    • Couldn't match expected type ‘OpenFileFlags’
                  with actual type ‘Maybe a0’
    • In the third argument of ‘openFd’, namely ‘Nothing’
      In the second argument of ‘($)’, namely
        ‘openFd "/" ReadOnly Nothing defaultFileFlags’
      In a stmt of a 'do' block:
        rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
   |
48 |        rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
   |                                               ^^^^^^^
[2 of 9] Compiling System.Unix.Crypt ( System/Unix/Crypt.hs, dist/build/System/Unix/Crypt.dyn_o )
[3 of 9] Compiling System.Unix.Directory ( System/Unix/Directory.hs, dist/build/System/Unix/Directory.dyn_o )
[4 of 9] Compiling System.Unix.FilePath ( dist/build/System/Unix/FilePath.hs, dist/build/System/Unix/FilePath.dyn_o )
[5 of 9] Compiling System.Unix.Files ( System/Unix/Files.hs, dist/build/System/Unix/Files.dyn_o )
[6 of 9] Compiling System.Unix.KillByCwd ( System/Unix/KillByCwd.hs, dist/build/System/Unix/KillByCwd.dyn_o )
[7 of 9] Compiling System.Unix.Misc ( System/Unix/Misc.hs, dist/build/System/Unix/Misc.dyn_o )
[8 of 9] Compiling System.Unix.Mount ( System/Unix/Mount.hs, dist/build/System/Unix/Mount.dyn_o )
[9 of 9] Compiling System.Unix.SpecialDevice ( System/Unix/SpecialDevice.hs, dist/build/System/Unix/SpecialDevice.dyn_o )
```

There seems to be a change in  `openFd` such that the `Nothing` becomes redundant.

### System information
OS: Arch Linux
Linux: 6.6.6.arch1-1
GHC: 9.6.3
base: 4.18.1.0
unixutils: 1.54.2

--- a/System/Unix/Chroot.hs
+++ b/System/Unix/Chroot.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE CPP #-}
 -- | This module, except for useEnv, is copied from the build-env package.
 module System.Unix.Chroot
     ( fchroot
@@ -45,7 +46,11 @@ chroot fp = withCString fp $ \cfp -> throwErrnoIfMinus1_ "chroot" (c_chroot cfp)
 fchroot :: (MonadIO m, MonadMask m) => FilePath -> m a -> m a
 fchroot path action =
     do origWd <- liftIO $ getWorkingDirectory
+#if MIN_VERSION_unix(2,8,0)
        rootFd <- liftIO $ openFd "/" ReadOnly defaultFileFlags
+#else
+       rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
+#endif
        liftIO $ chroot path
        liftIO $ changeWorkingDirectory "/"
        action `finally` (liftIO $ breakFree origWd rootFd)

--- a/System/Unix/Chroot.hs
+++ b/System/Unix/Chroot.hs
@@ -45,7 +45,7 @@ chroot fp = withCString fp $ \cfp -> throwErrnoIfMinus1_ "chroot" (c_chroot cfp)
 fchroot :: (MonadIO m, MonadMask m) => FilePath -> m a -> m a
 fchroot path action =
     do origWd <- liftIO $ getWorkingDirectory
-       rootFd <- liftIO $ openFd "/" ReadOnly Nothing defaultFileFlags
+       rootFd <- liftIO $ openFd "/" ReadOnly defaultFileFlags
        liftIO $ chroot path
        liftIO $ changeWorkingDirectory "/"
        action `finally` (liftIO $ breakFree origWd rootFd)

